### PR TITLE
Add certificate reconciler from knative/serving to knative/net-certmanager, preserving history

### PR DESCRIPTION
**Pull request by ZhiminXiang**
We will follow up with additional directory renames / moves in a separate PR.

part of https://github.com/knative/serving/issues/7306

### CODE REVIEW

File: pkg/reconciler/certificate/certificate.go [hunk 1]
```diff
@@ -0,0 +1,238 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline pkg/reconciler/certificate/certificate.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/certificate_test.go [hunk 2]
```diff
@@ -0,0 +1,610 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline pkg/reconciler/certificate/certificate_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/config/cert_manager.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline pkg/reconciler/certificate/config/cert_manager.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/config/cert_manager_test.go [hunk 4]
```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline pkg/reconciler/certificate/config/cert_manager_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/config/doc.go [hunk 5]
```diff
@@ -0,0 +1,20 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline pkg/reconciler/certificate/config/doc.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/config/store.go [hunk 6]
```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2019 The Knative Authors.
```
Comment (mattmoor) [inline pkg/reconciler/certificate/config/store.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors.
```

File: pkg/reconciler/certificate/config/store_test.go [hunk 7]
```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline pkg/reconciler/certificate/config/store_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/controller.go [hunk 8]
```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline pkg/reconciler/certificate/controller.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/resources/cert_manager_certificate.go [hunk 9]
```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline pkg/reconciler/certificate/resources/cert_manager_certificate.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/resources/cert_manager_certificate_test.go [hunk 10]
```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2019 The Knative Authors
```
Comment (mattmoor) [inline pkg/reconciler/certificate/resources/cert_manager_certificate_test.go]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/certificate.go [hunk 11]
```diff
@@ -0,0 +1,238 @@
...
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/logging"
+	pkgreconciler "knative.dev/pkg/reconciler"
+	"knative.dev/pkg/tracker"
+	"knative.dev/serving/pkg/apis/networking/v1alpha1"
+	certmanagerclientset "knative.dev/serving/pkg/client/certmanager/clientset/versioned"
+	acmelisters "knative.dev/serving/pkg/client/certmanager/listers/acme/v1alpha2"
+	certmanagerlisters "knative.dev/serving/pkg/client/certmanager/listers/certmanager/v1alpha2"
+	"knative.dev/serving/pkg/reconciler/certificate/config"
+	"knative.dev/serving/pkg/reconciler/certificate/resources"
```
Comment (mattmoor) [inline pkg/reconciler/certificate/certificate.go]: I believe these need to change to `knative.dev/net-certmanager/pkg/reconciler/certificate/...`

**Comment by mattmoor** [comment]
/hold

Placing a hold on this as I'll need to merge it manually to preserve the commit history.

File: pkg/reconciler/certificate/certificate.go (hunk 1 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/certificate.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/certificate_test.go [hunk 12]
```diff
@@ -0,0 +1,611 @@
...
```
Comment (mattmoor) [inline pkg/reconciler/certificate/certificate_test.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/config/cert_manager.go (hunk 3 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/config/cert_manager.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/config/cert_manager_test.go (hunk 4 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/config/cert_manager_test.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/config/doc.go (hunk 5 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/config/doc.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/config/store.go (hunk 6 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/config/store.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors.
```

File: pkg/reconciler/certificate/config/store_test.go (hunk 7 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/config/store_test.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/controller.go (hunk 8 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/controller.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/resources/cert_manager_certificate.go (hunk 9 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/resources/cert_manager_certificate.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/resources/cert_manager_certificate_test.go (hunk 10 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/resources/cert_manager_certificate_test.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/certificate.go (hunk 1 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/certificate.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/certificate_test.go (hunk 12 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/certificate_test.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/config/cert_manager.go (hunk 3 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/config/cert_manager.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/config/cert_manager_test.go (hunk 4 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/config/cert_manager_test.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/config/doc.go (hunk 5 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/config/doc.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/config/store.go (hunk 6 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/config/store.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors.
```

File: pkg/reconciler/certificate/config/store_test.go (hunk 7 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/config/store_test.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/controller.go (hunk 8 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/controller.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/resources/cert_manager_certificate.go (hunk 9 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/resources/cert_manager_certificate.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/resources/cert_manager_certificate_test.go (hunk 10 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/resources/cert_manager_certificate_test.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/certificate.go (hunk 1 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/certificate.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/certificate_test.go (hunk 12 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/certificate_test.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/config/cert_manager.go (hunk 3 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/config/cert_manager.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/config/cert_manager_test.go (hunk 4 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/config/cert_manager_test.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/config/doc.go (hunk 5 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/config/doc.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/config/store.go (hunk 6 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/config/store.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors.
```

File: pkg/reconciler/certificate/config/store_test.go (hunk 7 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/config/store_test.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/controller.go (hunk 8 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/controller.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/resources/cert_manager_certificate.go (hunk 9 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/resources/cert_manager_certificate.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

File: pkg/reconciler/certificate/resources/cert_manager_certificate_test.go (hunk 10 above)
Comment (mattmoor) [inline pkg/reconciler/certificate/resources/cert_manager_certificate_test.go:2]: Incorrect copyright year:
```suggestion
Copyright 2020 The Knative Authors
```

**Comment by ZhiminXiang** [comment]
Chatted with @chaodaiG offline. the test pull-knative-net-certmanager-go-coverage  should be recovered after this PR is merged.
So I think this PR is ready for review.

**Comment by mattmoor** [comment]
/lgtm
/approve
/hold

I am HOLDing this so that tide does not merge it (so we can preserve the commit history), however, @tcnghia needs to fix the CLA first by adding the `cla: yes` label to this PR (I cannot do this).

**Pull request merged**